test/functional/test_node_ports.py

```diff
@@ -144,3 +145,303 @@ def test_start_node1_then_start_node2_no_conflict_ports(self):
         assert "test1" in ipmi_result and "test2" in ipmi_result
         assert "test1-node" in qemu_result and "test2-node" in qemu_result
 
+    def test_start_node1_then_start_node2_bmc_connection_port(self):
+        """
+        Port test: after node1 start, if node2 also use the same bmc_connection_port to start, it won't start
+        """
+        # start node test1
+        self.node_info['name'] = "test1"
+        node1 = model.CNode(self.node_info)
+        node1.init()
+        node1.precheck()
+        node1.start()
+        time.sleep(2)
+
+        socat_result = run_command(PS_SOCAT, True,
+                                   subprocess.PIPE, subprocess.PIPE)[1]
+        ipmi_result = run_command(PS_IPMI, True,
+                                  subprocess.PIPE, subprocess.PIPE)[1]
+        qemu_result = run_command(PS_QEMU, True,
+                                  subprocess.PIPE, subprocess.PIPE)[1]
+        # check if node test1 is running
+        assert "test1" in socat_result
+        assert "test1" in ipmi_result
+        assert "test1-node" in qemu_result
+
+        self.node_info_2['name'] = "test2"
+        # modify node configuration to resolve port conflict
+        self.node_info_2['ipmi_console_ssh'] = 9301
+        self.node_info_2['ipmi_console_port'] = 9001
+        #self.node_info_2['bmc_connection_port'] = 9101
+        self.node_info_2['compute']['vnc_display'] = 2
+        self.node_info_2['compute']['monitor'] = {
+            'mode': 'readline',
+            'chardev': {
+                'backend': 'socket',
+                'host': '127.0.0.1',
+                'port': 2346,
+                'server': True,
+                'wait': False
+            }
+        }
+
+        try:
+            node2 = model.CNode(self.node_info_2)
+            node2.init()
+            node2.precheck()
+            node2.start()
```

Comment:
This exception should be raised in `precheck()`, so `start()` can be omitted here. 
Then I see your below tests to check there should be no active process.

Good job!